src/streamlink/plugins/teamliquid.py

```diff
@@ -0,0 +1,33 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import http
+
+
+_url_re = re.compile(r'''https?://(?:www\.)?teamliquid\.net/video/streams/''')
+_afreecaRe = re.compile('View on Afreeca')
+_twitchRe = re.compile('View on Twitch.tv')
+
+
+class Teamliquid(Plugin):
+    @classmethod
+    def can_handle_url(cls, url):
+        return _url_re.match(url)
+
+    def _get_streams(self):
+        res = http.get(self.url)
+        
+        afreeca=_afreecaRe.findall(res.text)
```

Comment:
Rather than searching for the specific strings `View on Afreeca` or `View on Twitch.tv` you might be better searching for any URL that is directly before the string `View on`. That way you can find other embedded streams, I noticed that there are some goodgame.ru streams too. You could use a regex like:
```python
stream_address_re = re.compile(r'''href\s*=\s*"([^"]*?)"[^<>]+?>\s*View\s+on\s+''', re.IGNORECASE | re.MULTILINE)
```
Then extract the matched URL (if any), and pass it to `self.session.streams`;
```python
stream_url_match = stream_address_re.search(res.text)
if stream_url_match:
    stream_url = stream_url_match.group(1)
    self.logger.info("Attempting to play streams from {0}", stream_url)
    return self.session.streams(stream_url)
```

That way you will be able to support any embedded streams that streamlink supports, not just twitch/afreeca.